src/event-log.c

```diff
@@ -0,0 +1,405 @@
+#include <glib.h>
+#include <gio/gio.h>
+#include <glib/gstdio.h>
+#if ENABLE_JSON
+#include <json-glib/json-glib.h>
+#endif
+
+#include "context.h"
+#include "event-log.h"
+
+void r_event_log_message(const gchar *type, const gchar *message, ...)
+{
+	va_list list;
+	g_autofree gchar *formatted = NULL;
+
+	g_return_if_fail(message);
+
+	va_start(list, message);
+	formatted = g_strdup_vprintf(message, list);
+	va_end(list);
+
+	g_log_structured(R_EVENT_LOG_DOMAIN, G_LOG_LEVEL_MESSAGE,
+			"RAUC_EVENT_TYPE", type,
+			"MESSAGE", "%s", formatted);
+}
+
+void r_event_log_free_logger(REventLogger *logger)
+{
+	if (!logger)
+		return;
+
+	g_clear_pointer(&logger->name, g_free);
+	g_clear_pointer(&logger->filename, g_free);
+	g_clear_pointer(&logger->logstream, g_object_unref);
+
+
+	g_free(logger);
+
+	return;
+}
+
+/**
+ * Log formatter for json output.
+ *
+ * Simply converts all log fields into a json object.
+ *
+ * | {
+ * |   "date": "2023-06-14T21:15:41Z"
+ * |   "MESSAGE" : "Booted into rootfs.0 (A)",
+ * |   "MESSAGE_ID" : "e60e0add-d345-4cb8-b796-eae0d497af96",
```

Comment:
Different in which way? It's just an example with two entries. Could be shortened to one.

Reply:
Ok, misunderstood you. So the reason for having "date" as it is is just that i wasn't sure if its the right name and I first of all wanted to differentiate it from the other entries since it is the only one generated in the log function itself.

Thus could be changed to `"DATE"` or `"TIMESTAMP"`.

Reply:
Is it useful to save some space in the JSON output? If so, perhaps `"TS"` and `"MSG(_ID)"`?

Reply:
I used the `journalctl -o json` as a reference which also just plainly prints out the internal names of structured logging.